.github/workflows/CI-build-test.yml

```diff
@@ -1,69 +1,65 @@
-name: CI-build-test
+name: Build-and-Test
 
-run-name: "Workflow performing CI steps: build, unit testing, integration testing"
+run-name: "Workflow CI/CD Steps: build, unit and integration testing"
 
 on:
   workflow_call:
 
 env:
-  RUST_BACKTRACE: 1
+  IMAGE_NAME: zencash/sc-ci-base
+  IMAGE_TAG: noble_rust-stable_latest
+  DOCKER_COMPOSE_CMD: "docker compose -f ${GITHUB_WORKSPACE}/ci/docker-compose.yml run --rm cargo-container"
+  DOCKER_BUILD_DIR: /build
+  DOCKER_CARGO_HOME: /tmp/.cargo
 
 jobs:
-  build-test:
-    strategy:
-      matrix:
-        os: [ubuntu-latest]
-        # os: [ubuntu-latest, windows-latest, macos-latest]
-    runs-on: ${{ matrix.os }}
+  build-and-test:
+    runs-on: ubuntu-latest
     steps:
-      - name: Checkout branch
+      - name: Git checkout
         uses: actions/checkout@v4
+
       - name: Set up cargo cache
         uses: actions/cache@v4
-        continue-on-error: false
+        if: ${{ !env.ACT }}
+        continue-on-error: true
         with:
           path: |
             ~/.cargo/bin/
             ~/.cargo/registry/index/
             ~/.cargo/registry/cache/
             ~/.cargo/git/db/
             target/
-          key: ${{ runner.os }}-build-test-cargo-${{ hashFiles('**/Cargo.lock') }}
-          restore-keys: ${{ runner.os }}-build-test-cargo-
-      - name: Install deps
-      # required by sc-network-sync
-      # required by nh-runtime
-      # required by nh-runtime
+          key: build-test-cargo-${{ hashFiles('**/Cargo.lock') }}
+
+      - name: Set docker env vars
         run: |
-            sudo apt-get install protobuf-compiler
-            rustup target add wasm32-unknown-unknown
-            rustup component add rust-src
-      - name: Check
-        run: cargo check --all-features --all-targets --release
```

Comment:
It's not related to your PR (already present) but please take this chance for removing `--release` here (we are already using `--all-targets`)

Reply:
Done